crates/solidity/outputs/cargo/crate/src/extensions/semver/mod.rs

```diff
@@ -0,0 +1,463 @@
+pub mod parser;
+pub mod tests;
+
+use std::fmt::Display;
+
+use semver::Version;
+
+use crate::cst::NonterminalKind;
+use crate::utils::LanguageFacts;
+
+/// Parse the version pragmas in the given Solidity source code and return a list of language
+/// versions that can fulfill those requirements.
+pub fn infer_language_version(src: &str) -> Vec<Version> {
```

Comment:
I see both sub-modules and other types here are `pub`. Is that needed?
I think only the main `fn infer_language_version()` is actually used.